src/word.rs

```diff
@@ -717,6 +717,157 @@ impl<'a> UWordBounds<'a> {
     }
 }
 
+pub struct AsciiWordBoundIter<'a> {
+    rest: &'a str,
+    offset: usize,
+}
+
+impl<'a> AsciiWordBoundIter<'a> {
+    pub fn new(s: &'a str) -> Self {
+        AsciiWordBoundIter { rest: s, offset: 0 }
+    }
+
+    #[inline]
+    fn is_core(b: u8) -> bool {
+        b.is_ascii_alphanumeric() || b == b'_'
+    }
+
+    #[inline]
+    fn is_infix(b: u8, prev: u8, next: u8) -> bool {
+        match b {
+            // numeric separators
+            b'.' | b',' | b';' | b'\'' if prev.is_ascii_digit() && next.is_ascii_digit() => true,
+            // apostrophe in contractions
+            b'\'' if prev.is_ascii_alphabetic() && next.is_ascii_alphabetic() => true,
+            // dot/colon inside letters
+            b'.' | b':' if prev.is_ascii_alphabetic() && next.is_ascii_alphabetic() => true,
+            _ => false,
+        }
+    }
+}
+
+impl<'a> Iterator for AsciiWordBoundIter<'a> {
```

Comment:
I'm worried about this code not matching the spec; it should have significantly more comments

Reply:
You'd be surprised, some of the segmentation APIs occasionally change around subtle ASCII things like punctuation norms. I think we had a change to line segmentation recently.